### PR TITLE
Openroad

### PR DESCRIPTION
**Pull request by osamahammad21**
This pull request includes the following changes:

1. Adding python unit tests to db objects (check the readme in unitTestsPython dir)
2. Bug fixes:
    - Hierarchical destruction for dbRegions and dbBlocks
    - Improved the implementation of computeBBox to put into consideration continuous changes in the design
    - Applied needed changes to python swig interface

### CODE REVIEW

File: src/db/dbBPin.cpp [hunk 1]
```diff
@@ -188,6 +188,8 @@ void dbBPin::setPlacementStatus(dbPlacementStatus status)
 {
   _dbBPin* bpin        = (_dbBPin*) this;
   bpin->_flags._status = status.getValue();
+  _dbBlock* block = (_dbBlock*) bpin->getOwner();
+  block->_flags._valid_bbox=0;
```
Comment (maliberty) [inline src/db/dbBPin.cpp:192]: = false  (its really a boolean field)

File: src/db/dbInst.cpp [hunk 2]
```diff
@@ -449,6 +449,7 @@ void dbInst::setOrigin(int x, int y)
   for (cbitr = block->_callbacks.begin(); cbitr != block->_callbacks.end();
        ++cbitr)
     (**cbitr)().inDbMoveInst(this);
+  block->_flags._valid_bbox=0;
```
Comment (maliberty) [inline src/db/dbInst.cpp:452]: =false

File: src/db/dbInst.cpp [hunk 3]
```diff
@@ -512,6 +515,7 @@ void dbInst::setOrient(dbOrientType orient)
   for (cbitr = block->_callbacks.begin(); cbitr != block->_callbacks.end();
        ++cbitr)
     (**cbitr)().inDbMoveInst(this);
+  block->_flags._valid_bbox = 0;
```
Comment (maliberty) [inline src/db/dbInst.cpp:518]: ditto

File: src/db/dbInst.cpp [hunk 4]
```diff
@@ -526,7 +530,8 @@ void dbInst::setPlacementStatus(dbPlacementStatus status)
   //_dbBlock * block = (_dbBlock *) getOwner();
   // uint prev_flags = flagsToUInt(inst);
   inst->_flags._status = status.getValue();
-
+  _dbBlock*  block =(_dbBlock*) getBlock();
+  block->_flags._valid_bbox = 0;
```
Comment (maliberty) [inline src/db/dbInst.cpp:534]: ditto

File: tests/cpp/test.cpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#include "db.h"
+#include <iostream>
+using namespace odb;
+int main(int argc, char *argv[])
+{
+    dbDatabase* db = dbDatabase::create();
+    std::cout<<"2"<<std::endl;
```
Comment (maliberty) [inline tests/cpp/test.cpp:7]: What is this testing?

File: src/db/dbBPin.cpp (hunk 1 above)
Comment (osamahammad21) [reply src/db/dbBPin.cpp:192]: actually it is of uint type and has been treated as such in the whole project
Screenshot from 2020-08-10 19-02-04

**Pull request merged**